compute_sdk/tests/unit/test_executor.py

```diff
@@ -635,6 +683,37 @@ def _mock_max(*a, **k):
         assert found_tg_uuid == expected
 
 
+def test_task_submit_handles_multiple_user_endpoint_configs(
+    gc_executor: tuple[mock.MagicMock, Executor]
+):
+    gcc, gce = gc_executor
+    gce.endpoint_id = uuid.uuid4()
+
+    func_uuid_str = str(uuid.uuid4())
+    tg_uuid_str = str(uuid.uuid4())
+    gcc.register_function.return_value = func_uuid_str
+    gcc.batch_run.side_effect = (
+        ({"task_group_id": tg_uuid_str, "tasks": {func_uuid_str: [str(uuid.uuid4())]}}),
+        ({"task_group_id": tg_uuid_str, "tasks": {func_uuid_str: [str(uuid.uuid4())]}}),
+    )
+
+    uep_config_1 = {"heartbeat": 10}
+    uep_config_2 = {"heartbeat": 20}
+    gce.user_endpoint_config = uep_config_1
+    gce.submit(noop)
+    gce.user_endpoint_config = uep_config_2
+    gce.submit(noop)
+
+    assert not gcc.create_batch.called, "Verify test setup"
```

Comment:
Glad this works, but ... how?  What prevents the backing thread from taking one or both of the tasks before line 707 is hit?

Reply:
I intended to include a thread event to temporarily block the loop (as we do [here](https://github.com/funcx-faas/funcX/blob/ae2f3499a54e54cdd8c03c32a585bb8ae023f349/compute_sdk/tests/unit/test_executor.py#L609-L615)), but looks like I mistakenly left it out. I suppose the test passes because the timing is just right. I'll add it in a fresh commit.